### DISCUSSION TITLE
Decouple multiple element type selection from RelatedToConditionRule

### DISCUSSION THREAD

**Discussion by rynpsc** [craftcms/cms#11066] [Ideas] [answered]
The "Related To" condition rule allows the user to choose from any element type via `RelatedToConditionRule` which inherits from `BaseElementSelectConditionRule`.

Could a `multiple` property be added to `BaseElementSelectConditionRule` and then some of the logic within `RelatedToConditionRule` be moved to `BaseElementSelectConditionRule` instead? This would allow easier reuse of the multiple element selection for other purposes other than for relations.

For example in Craft Commerce the "Matching Items" for a discount could be moved to use the condition builder allowing things other than categories to be selected, or in a plugin which collects "comments" for any Element, comments could be filtered on the element index based on element.

This would also be relevant to https://github.com/craftcms/commerce/discussions/2796

I'm happy to make a PR.

**Comment by brandonkelly** [comment]
This would be a breaking change at this point, as it would involve renaming `BaseElementSelectConditionRule::getElementId()` / `setElementId()`, or at least changing their signatures. Will add it to the 5.0 list though.

**Comment by brandonkelly** [comment, marked as answer]
We ended up adding support for multiple element selections in Craft 5.6 (6bf0270eae04ca5ed22bcad9fb6e294d04009dfd).
